### ISSUE TITLE
Pattern function could probably store compiled patterns.

### ISSUE THREAD

**Issue by cristovarghese** [go-openapi/validate#16]
While doing so some optimization for an application using swagger, I noticed patterns are being compiled repeatedly. This is wasting CPU cycles for garbage collection routines and also unnecessary repeated compilation. Is it possible to cache the the compiled patterns in some  map which can be re-used if necessary.

``` go
// Pattern validates a string against a regular expression
func Pattern(path, in, data, pattern string) *errors.Validation {
    re := regexp.MustCompile(pattern)
    if !re.MatchString(data) {
        return errors.FailedPattern(path, in, pattern)
    }
    return nil
}
```

**Comment by GlenDC** [comment]
PR #18 has been merged, showing there are quite a lot of places where patterns occur and regular expressions get compiled. We could definitely cache the results and errors.

The one and only approach I currently see being possible, is using a global somehow for this (in- or directly exposed).

Perhaps @casualjim has some better ideas regarding this suggestion.

**Comment by casualjim** [comment]
fixed by: go-openapi/validate#44
